### DISCUSSION TITLE
Search against array

### DISCUSSION THREAD

**Discussion by wind57** [provectus/kafka-ui#2581] [Q&A] [answered]
This is basically a follow-up of [this](https://github.com/provectus/kafka-ui/discussions/2579).

So, how does one provide a syntax for a search against an array?

Everything I tried so far ends up in 400 Bad Request. For example:

```
value.animals.each{ x -> x.name == 'monkey'}
```

**Comment by wind57** [comment]
If I do : 


```
def animals = value.animals
return animals.any(x -> x.name == 'monkey')
```

I get zero results.

And of course if I do this in a class:


```
class ParseJson {

    static void main(String[] args) {
        def jsonSlurper = new JsonSlurper()
        def value = jsonSlurper.parseText('''
{
  "animals": [
    {
      "name": "monkey"
    }
  ]
}
''')

        def animals = value.animals
        boolean  x = animals.any(x -> x.name == 'monkey')
        println(x)
    }

}
```

it works.

**Reply by Haarolean** [reply]
Thanks, I've opened up an issue.

**Comment by Haarolean** [comment]
Something like this should work
![image](https://user-images.githubusercontent.com/1494347/190286790-ebcad408-c40e-4681-856b-283630c38e37.png)

**Comment by wind57** [comment]
as said, this does _not_ work. I mean - yeah, from groovy it does, but not from the "smart filter"

**Reply by Haarolean** [reply]
Sorry, missed your other message.

**Reply by wind57** [reply]
No worries. Thank you for looking into it

**Reply by iliax** [reply, marked as answer]
@wind57 what image version do you use? We have expression escaping problem in current stable (latest) version, master should be fine. Also check my comment here (checked on master image) -> https://github.com/provectus/kafka-ui/discussions/2579#discussioncomment-3653299
